File: recursividade/java/src/Fibonacci.java
public class Fibonacci {

	public static void main(String[] args) {
        int result = fib(6);
        System.out.println("RESULTADO = " + result);
    }

    public static int fib(int n) {
        return fibTailRecursive(n, 0, 1);
    }

    public static int fibTailRecursive(int n, int a, int b) {
        if (n == 0) {
            return a;
        }
        if (n == 1) {
            return b;
        }
        return fibTailRecursive(n - 1, b, a + b);
    }
}

File: recursividade/python/fibonacci.py
def fib(n):
    return fib_tail_recursive(n, 0, 1)


def fib_tail_recursive(n, a, b):
    if n == 0:
        return a
    if n == 1:
        return b
    return fib_tail_recursive(n - 1, b, a + b)


result = fib(6)
print("RESULTADO =", result)
